src/status_im/chat/models/images.cljs

```diff
@@ -104,30 +104,65 @@
 
 (re-frame/reg-fx
  ::camera-roll-get-photos
- (fn [[num end-cursor]]
+ (fn [[num end-cursor album]]
    (permissions/request-permissions
     {:permissions [:read-external-storage]
      :on-allowed  (fn []
-                    (-> (if end-cursor
-                            (.getPhotos
-                             CameraRoll
-                             #js
-                              {:first      num
-                               :after      end-cursor
-                               :assetType  "Photos"
-                               :groupTypes "All"
-                               :include    (clj->js ["imageSize"])})
-                            (.getPhotos CameraRoll
-                                        #js
-                                         {:first      num
-                                          :assetType  "Photos"
-                                          :groupTypes "All"
-                                          :include    (clj->js ["imageSize"])}))
+                    (-> (.getPhotos
+                         CameraRoll
+                         #js
+                          {:first      num
+                           :after      end-cursor
+                           :assetType  "Photos"
+                           :groupTypes (if (= album (i18n/label :t/recent)) "All" "Albums")
+                           :groupName  (when (not= album (i18n/label :t/recent)) album)
+                           :include    (clj->js ["imageSize"])})
                         (.then #(let [response (types/js->clj %)]
                                   (re-frame/dispatch [:on-camera-roll-get-photos (:edges response)
                                                       (:page_info response) end-cursor])))
                         (.catch #(log/warn "could not get camera roll photos"))))})))
 
+(re-frame/reg-fx
+ ::camera-roll-get-albums
+ (fn []
+   (let [albums (atom [{:title :smart-albums :data []}
+                       {:title (i18n/label :t/my-albums) :data []}])]
+     ;; Get the "recent" album first
```

Comment:
maybe you could just put this in a named helper function to avoid the comment.
e.g 
```
(defn get-recent-album [...]
(->
       (.getPhotos CameraRoll
                   #js
                    {:first      1
                     :groupTypes "All"})
       (.then
        (fn [res]
          (let [recent-album {:title (i18n/label :t/recent)
                              :count "--"
                              :uri   (get-in (first (:edges (types/js->clj res))) [:node :image :uri])}]
            (swap! albums update-in [0 :data] conj recent-album))
            ```

Reply:
That would look better, but I need the rest of the `getAlbums` method to be nested inside the `.then` call from getting recent album (and the following `getPhotos` to be nested inside `getAlbums` `.then` call). 

I was trying to look for something like async/await in javascript. @ilmotta pointed me to this library "[Promesa](https://cljdoc.org/d/funcool/promesa/1.8.1/doc/promesa-promise-library-for-clojure-and-clojurescript#install)". @flexsurfer wdyt of this library, can I add it to the project?

Reply:
if its usable why not :) its clojure not js :)

Reply:
@flexsurfer hahaha okay :D

Reply:
I worked in codebases with lots of promises on the client side, and `promesa` was always useful and easy to learn :)

In the case of the mobile codebase I would say we have few usages of promises, because almost all async calls are made via the RPC effect, so the library value is diminished.

Reply:
@ilmotta there are actually lots of `.then` calls in the codebase. I thing it would generally be better/more readable if we start using async/await.

Reply:
It depends on what we mean by *lots*. I only found 43 usages, and most of the usages are in places that rarely change, like utility namespaces. But I have no concerns in bringing the library in :)

>I think it would generally be better/more readable if we start using async/await.

`promesa` will give us a couple useful macros (like `promesa.core/let`), but the majority of the library uses functions (not macros). Hence, most of the code will still treat promises as values and will require using `->` to thread over them. I'm just trying to lower your expectations a little bit because `async/await` is a completely different monster that tries to hide promises under a dubious syntax with lots of corner cases.

Reply:
Alright cool, I will leave it as it is for now, and get back to it sometime later.